### ISSUE TITLE
DDP + Pipeline parallelism

### ISSUE THREAD

**Issue by prathameshtd** [pytorch/torchtitan#636]
For fine tuning/training with `PP + DDP`, is there documentation or modification that can be done to achieve this using torchtitan? 

The following check in `parallelize_llama.py` was the point of error when trying the configuration on my end.
`if world_mesh.ndim > 1:
                raise RuntimeError("DDP has not supported > 1D parallelism")`

The use case I am imagining is: for a host with multiple GPUs that is responsible for a particular pipeline stage (part of model), as long as there is enough memory `DDP` might be a viable option.

**Comment by awgu** [comment]
I wonder what is the motivation to use PP with DDP instead of FSDP?

**Comment by prathameshtd** [comment]
The motivation is primarily research oriented in comparing the training costs with respect to different configurations.

**Comment by fegin** [comment]
If memory is enough, then why PP + DDP? If memory is a problem, should using FSDP be the first option before trying PP? 

Going back to your question, for DDP + PP, one major blocker I can think of is disabling gradient synchronization. This is implemented in DDP for gradient accumulation. But it requires to wrap for both DDP's forward and backward. How to elegantly integrate the API with the current PP implementation needs some investigation.

**Comment by fegin** [comment]
I'm going to close the issue. Please reopen if you have further questions.
